### PR TITLE
[NBS 1.0] Mark `backend-common` as deprecated

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Mark `@backstage/backend-common` as deprecated.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend-common/package.json [hunk 1]
```diff
@@ -151,5 +151,6 @@
       "optional": true
     }
   },
-  "configSchema": "config.d.ts"
+  "configSchema": "config.d.ts",
+  "deprecated": true
```
Comment (camilaibs) [inline packages/backend-common/package.json:155]: Should I revert this change and do this instead?
https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions

Comment (freben) [reply packages/backend-common/package.json:155]: yep let's do that

Comment (camilaibs) [reply packages/backend-common/package.json]: Done 🙏🏻

**Review by freben — APPROVED** [review]
